### ISSUE TITLE
unmap pos_inds and neg_inds in anchor_target_single

### ISSUE THREAD

**Issue by caiqi** [open-mmlab/mmdetection#1199]
In anchor_target_single https://github.com/open-mmlab/mmdetection/blob/master/mmdet/core/anchor/anchor_target.py#L151 , the labels, label_weights, bbox_targets and bbox_weights are unmapped to original anchors but pos_inds and neg_inds are not. So the returned inds does not match the returned targets. Would it be better to also unmap the inds. For example,

> pos_inds = unmap(unmap(torch.ones_like(pos_inds), anchors.shape[0], pos_inds), num_total_anchors, inside_flags).nonzero()
> neg_inds = unmap(unmap(torch.ones_like(neg_inds), anchors.shape[0], neg_inds), num_total_anchors, inside_flags).nonzero()

**Comment by hellock** [comment]
Thanks! We will check it.
